apps/api/src/report/report.service.ts

```diff
@@ -276,22 +287,60 @@ export class ReportService {
     const { to } = this.captureFromTo('', t);
     this.logger.log(`getAverageTimeOfMilestones: apply filter till ${to} date)`);
 
-    const entityManager = getManager();
-    const data = await entityManager.query(
-      this.reportUtilService.getAverageTimeOfMilestones(statuses, to),
-    );
-    this.logger.log(
-      `getAverageTimeOfMilestones: query completed a total of ${data.length} record returns`,
-    );
-    return data;
+    const licRecDurations = await getRepository(MilestoneDurationEntity).find({
+      where: [{ hired_date: LessThanOrEqual(to) }, { withdraw_date: LessThanOrEqual(to) }],
+    });
+
+    // the end date of immigration milestones should be limited by 'to' date instead of hired date
+    // because immigration stage comes after recruitment
+    const immigrationDurations = await getManager().query(`
```

Comment:
more so just brainstorming, but this query is duplicated in the `CREATE VIEW` as part of the `milestone_duration` too, with just the `to` date replacing the `CURRENT_DATE`.  Wonder if we can reduce that?  Just leaving this here  while I continue thinking

Reply:
That's what I wanted but I couldn't find a way.